CommunityPromotionScreen/Src/NewPromotionScreenbyDefault/Classes/NPSBDP_UIArmory_PromotionHero.uc

```diff
@@ -601,52 +601,80 @@ function bool CanPurchaseAbilityEx(int Rank, int Branch, name AbilityName, out s
 function int GetAbilityPointCost(int Rank, int Branch)
 {
 	local XComGameState_Unit UnitState;
+	local X2SoldierClassTemplate ClassTemplate;
 	local array<SoldierClassAbilityType> AbilityTree;
 	local bool bPowerfulAbility;
+	local bool bFinalRankAbility;
+	local bool bXCOMRowAbility;
+	local bool bAsResistanceHero;
 	local int AbilityRanks; //Rank is 0 indexed but AbilityRanks is not. This means a >= comparison requies no further adjustments
 	local int AbilityCost;
+	local XComLWTuple Tuple; // Tuple for Issue #10
 
 	UnitState = GetUnit();
+	ClassTemplate = UnitState.GetSoldierClassTemplate();
 	AbilityTree = UnitState.GetRankAbilities(Rank);	
+	AbilityRanks = GetAbilitiesPerRank(UnitState);
+	bPowerfulAbility = class'X2StrategyGameRulesetDataStructures'.default.PowerfulAbilities.Find(AbilityTree[Branch].AbilityName) != INDEX_NONE;
+	bFinalRankAbility = Rank >= ClassTemplate.GetMaxConfiguredRank() - 1;
```

Comment:
To be honest I'd prefer to guard the ClassTemplate behind a none check instead, but I suppose if a soldier is missing a soldier class template the user has bigger things to worry about than log warning, so yeah. But let's not always do this, since long expressions can get hard to read (not in this case, obviously).